src/bgw_policy/drop_chunks.c

```diff
@@ -18,17 +19,65 @@
 #include "hypertable.h"
 #include "bgw/job.h"
 #include "scan_iterator.h"
+#include "bgw_policy/interval.h"
 
 static ScanTupleResult
 bgw_policy_drop_chunks_tuple_found(TupleInfo *ti, void *const data)
 {
 	BgwPolicyDropChunks **policy = data;
-
-	*policy = STRUCT_FROM_TUPLE(ti->tuple,
-								ti->mctx,
-								BgwPolicyDropChunks,
-								FormData_bgw_policy_drop_chunks);
-
+	bool nulls[Natts_bgw_policy_drop_chunks];
+	Datum values[Natts_bgw_policy_drop_chunks];
+	bool nulls_older_than[Natts_ts_interval];
+	Datum values_older_than[Natts_ts_interval];
+
+	/* cannot use STRUCT_FROM_TUPLE because some fields in the result may be NULLs*/
```

Comment:
```sql
is_time_interval        BOOLEAN,
time_interval	        INTERVAL,
integer_interval        BIGINT
```
in postgres sql I cannot add custom(e.g. NOT NULL) constraints on composite type fields.
I could still add a validation function and enforce that for example `is_time_interval` is never null.

That said, when `is_time_interval` is false, `time_interval` must be null.
Maybe I still get use `STRUCT_FROM_TUPLE` because sometimes postgres does not compress null fields out but I think doing so will rely on some internal postgres implementation.

Reply:
resolved in a call, you are right, nothing in `bgw_policy_drop_chunks` table can be null so will change back to `STRUCT_FROM_TUPLE`

Reply:
DISCUSS: I later noticed that if I leave older_than as deformed struct inside the FormData_bgw_policy_drop_chunks instead of a Datum, then I cannot just use STRUCT_FROM_TUPLE.

The reason to leave older_than deformed is to make the later code using older_than simpler

Reply:
Also, I tried converting it back to `STRUCT_FROM_TUPLE` just to have the code in case we decide to proceed with that approach. 

I did not manage to get the catalog.c FormData_bgw_policy_drop_chunks struct right. I am not sure how the custom row type must be stored and was unable to find any examples of that being done in Timescale or postgres source code.
It looks like `STRUCT_FROM_TUPLE` expects the struct we are copying into be 55 bytes but if I assume the row type is stored as datum, struct size is 24 and if I assume it is stored as a struct with all of its fields, it is 48 bytes